src/legacy/ui/public/saved_objects/types.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { SearchSource } from 'ui/courier';
+import { SavedObjectAttributes, SavedObjectReference } from 'kibana/server';
+import { IndexPattern } from '../../../core_plugins/data/public';
+
+export interface SavedObject {
+  _serialize: () => { attributes: SavedObjectAttributes; references: SavedObjectReference[] };
+  _source: Record<string, unknown>;
+  applyESResp: (resp: EsResponse) => any;
+  applyEsResp: () => any;
+  copyOnSave: boolean;
+  creationOpts: (opts: CreationOpts) => Record<string, unknown>;
+  defaults: any;
+  delete?: () => Promise<{}>;
+  destroy?: () => void;
+  getDisplayName: () => string;
+  getEsType: () => string;
+  getFullPath: () => string;
+  hydrateIndexPattern?: (id?: string) => angular.IPromise<null | void>;
```

Comment:
Does this definitely need to be `angular.IPromise` or can it be a native promise? From looking at the code, it looks like `hydrateIndexPattern` is only returning native promises, but I might be missing something.

Reply:
Currently migrating all to native promise where possible that's why the functional tests are failing

Reply:
where it's possible, it's migrated to native promise